src/pytestqt/wait_signal.py

```diff
@@ -43,11 +43,13 @@ def wait(self):
             return
         if self.timeout is None and not self._signals:
             raise ValueError("No signals or timeout specified.")
-        if self._timer is not None:
-            self._timer.timeout.connect(self._quit_loop_by_timeout)
-            self._timer.start()
 
         if self.timeout != 0:
+            if self.timeout is not None:
```

Comment:
Probably makes sense to unify these 2 ifs?

Reply:
Probably we can drop the `if self.timeout != 0` too btw, I would trust Qt even does that already.

Reply:
There's a `qt_api.exec(self._loop)` below which (at least for keeping the old logic) should be run if the timeout is `None` (= infinite), but should not be run if the timeout is 0 (docs: "If the timeout is set to `0`, it’s expected that the signal arrives directly in the code inside the with `qtbot.waitSignal(...):` block.").

I noticed we *don't* have the same handling in `CallbackBlocker`, which might need a review to make sure it all makes sense and is consistent somehow, but that seems out of scope for this PR.

Reply:
Sure thing, thanks!